docs/recipes/style-guide.md

```diff
@@ -45,3 +45,173 @@ Unit tests for the state should be named `my-state-name.state.spec.ts`
 
 Actions should NOT deal with view related operations (i.e. showing popups/etc). Use the action
 stream to handle these types of operations
+
+### Avoid Saving Class Based Instances in Your State
+
+The objects stored in your state should be immutable and should support serialization and deserialization. It is therefore recommended to store pure object literals in your state. Class based instances are not trivial to serialize and deserialize, and also are generally focused on encapsulating internals and mutating internal state through exposed operations. This does not match the requirement for the data stored in state.
+
+This also applies to the usage of data collections such as Set, Map, WeakMap, WeakSet, etc. Since they are not amenable to deserialization and cannot easily be presented for normalization.
+
+#### Avoid
+
+```ts
+export class Todo {
+  constructor(public title: string, private complete: boolean = false) {}
+  public get isCompleted(): boolean {
```

Comment:
What is this for?

Reply:
For example, I saw when people wrote such code

Reply:
Sorry, I've never seen it. We should not encourage people to write such code, as our documentation has authority. Let's stick wi the single `public isCompleted = false` property.

Reply:
done